lib/plugins/package/lib/zipService.js

```diff
@@ -226,28 +226,31 @@ function excludeNodeDevDependencies(servicePath) {
               .filter(item => item.length > 0 && item.match(nodeModulesRegex))
               .reduce((globs, item) => {
                 const packagePath = path.join(servicePath, item, 'package.json');
-                return fs.readFileAsync(packagePath, 'utf-8').then(packageJsonFile => {
-                  const lastIndex = item.lastIndexOf(path.sep) + 1;
-                  const moduleName = item.substr(lastIndex);
-                  const modulePath = item.substr(0, lastIndex);
-
-                  const packageJson = JSON.parse(packageJsonFile);
-                  const bin = packageJson.bin;
-
-                  const baseGlobs = [path.join(item, '**')];
-
-                  // NOTE: pkg.bin can be object, string, or undefined
-                  if (typeof bin === 'object') {
-                    _.each(_.keys(bin), executable => {
-                      baseGlobs.push(path.join(modulePath, '.bin', executable));
-                    });
-                    // only 1 executable with same name as lib
-                  } else if (typeof bin === 'string') {
-                    baseGlobs.push(path.join(modulePath, '.bin', moduleName));
-                  }
-
-                  return globs.concat(baseGlobs);
-                });
+                return fs
+                  .readFileAsync(packagePath, 'utf-8')
+                  .then(packageJsonFile => {
+                    const lastIndex = item.lastIndexOf(path.sep) + 1;
+                    const moduleName = item.substr(lastIndex);
+                    const modulePath = item.substr(0, lastIndex);
+
+                    const packageJson = JSON.parse(packageJsonFile);
+                    const bin = packageJson.bin;
+
+                    const baseGlobs = [path.join(item, '**')];
+
+                    // NOTE: pkg.bin can be object, string, or undefined
+                    if (typeof bin === 'object') {
+                      _.each(_.keys(bin), executable => {
+                        baseGlobs.push(path.join(modulePath, '.bin', executable));
+                      });
+                      // only 1 executable with same name as lib
+                    } else if (typeof bin === 'string') {
+                      baseGlobs.push(path.join(modulePath, '.bin', moduleName));
+                    }
+
+                    return globs.concat(baseGlobs);
+                  })
+                  .catch(() => globs);
```

Comment:
To reflect properly if/else, let's pass this `catch` callback as second argument to previous `then` (otherwise it also handles errors which may happen in first `then` callback and that can impose nasty, difficult to debug issues)

Reply:
@medikoo Not really sure if I understood you correctly. If I'm not mistaken, the `catch` block I added only handles the error from `readFileAsync` it does not handle errors of any previous `then` blocks.

Reply:
> it does not handle errors of any previous then blocks.

@darko1979 it handles. To ensure it'll only handle error from `readFileAsync` it needs to be attached directly to promise returned by `readFileAsync` (and to result of `then()` in which we process it's result)

Reply:
@medikoo I understand now, I corrected it by your suggestion